SimTKcommon/Geometry/include/SimTKcommon/internal/DecorativeGeometry.h

```diff
@@ -596,6 +596,79 @@ class SimTK_SimTKCOMMON_EXPORT DecorativeMeshFile : public DecorativeGeometry {
 };
 
 
+/** This defines a displayable torus  . **/
+class SimTK_SimTKCOMMON_EXPORT DecorativeTorus : public DecorativeGeometry {
+public:
+    explicit DecorativeTorus(Real inner=0.1, Real outer=0.5);
+
+    DecorativeTorus& setInnerRadius(Real);
+    DecorativeTorus& setOuterRadius(Real);
+    Real getInnerRadius() const;
+    Real getOuterRadius() const;
+
+    // Retain the derived type when setting generic geometry options.
+    DecorativeTorus& setBodyId(int b)          { DecorativeGeometry::setBodyId(b);        return *this; }
+    DecorativeTorus& setIndexOnBody(int x)     { DecorativeGeometry::setIndexOnBody(x);   return *this; }
+    DecorativeTorus& setUserRef(void* p)       { DecorativeGeometry::setUserRef(p);       return *this; }
+    DecorativeTorus& setTransform(const Transform& X_BD) { DecorativeGeometry::setTransform(X_BD); return *this; }
+    DecorativeTorus& setResolution(Real r)     { DecorativeGeometry::setResolution(r);    return *this; }
+    DecorativeTorus& setScaleFactors(const Vec3& s) { DecorativeGeometry::setScaleFactors(s); return *this; }
+    DecorativeTorus& setColor(const Vec3& rgb) { DecorativeGeometry::setColor(rgb);       return *this; }
+    DecorativeTorus& setOpacity(Real o)        { DecorativeGeometry::setOpacity(o);       return *this; }
+    DecorativeTorus& setLineThickness(Real t)  { DecorativeGeometry::setLineThickness(t); return *this; }
+    DecorativeTorus& setRepresentation(const Representation& r)
+    {
+        DecorativeGeometry::setRepresentation(r); return *this;
+    }
+
+    SimTK_PIMPL_DOWNCAST(DecorativeTorus, DecorativeGeometry);
+private:
+    class DecorativeTorusRep& updRep();
+    const DecorativeTorusRep& getRep() const;
+};
+
+
+/** An arrow with origin point, direction and length. Note that the actual 
+placement can be changed by the parent class transform & scale; here we are 
+just generating the initial arrow in the geometry object's local frame.
+
+There is a default constructor for this object but it is not much
+use unless followed by point, direction and length specifications. By default 
+we produce an arrow going from (0,0,0) in direction (1,1,1) of length 1 just 
+so it will show up if you forget to set it to something meaningful. Having a 
+default constructor allows us to have arrays of these objects. **/
+class SimTK_SimTKCOMMON_EXPORT DecorativeArrow : public DecorativeGeometry {
+public:
+    explicit DecorativeArrow(const Vec3& p1 = Vec3(0), const Vec3& dir=Vec3(1), Real length=1.0); // Arrow 
```

Comment:
> ... use unit-vector as suggested and tackle the wrapping issue sometime later. Do you have a preference?

I do want the main constructor to use `UnitVec3`, but it would be fine to have a "convenience" constructor that takes a `Vec3` if that helps wrapping. Its entire implementation could be this in the header file:
```c++
DecorativeArrow(const Vec3& p1, const Vec3& dir, Real length=1.0) : DecorativeArrow(p1,UnitVec3(dir),length) {}
```